src/NServiceBus.Core/Performance/MessageProcessingOptimizations/MessageProcessingOptimizationExtensions.cs

```diff
@@ -15,15 +17,7 @@ public static void LimitMessageProcessingConcurrencyTo(this EndpointConfiguratio
             Guard.AgainstNull(nameof(config), config);
             Guard.AgainstNegativeAndZero(nameof(maxConcurrency), maxConcurrency);
 
-            config.Settings.Set(new ConcurrencyLimit
-            {
-                MaxValue = maxConcurrency
-            });
-        }
-
-        internal class ConcurrencyLimit
-        {
-            public int MaxValue { get; set; }
+            config.Settings.Get<ReceiveComponent.Settings>().PushRuntimeSettings = new PushRuntimeSettings(maxConcurrency);
```

Comment:
this works for now as cocnurrency is the only property in `PushRuntimeSettings` but this might become an issue if there is something added because this would override other instances of the class